pkg/cli/jobs.go

```diff
@@ -0,0 +1,88 @@
+package cli
+
+import (
+	"context"
+
+	cli "github.com/acorn-io/runtime/pkg/cli/builder"
+	"github.com/acorn-io/runtime/pkg/cli/builder/table"
+	"github.com/acorn-io/runtime/pkg/client"
+	"github.com/acorn-io/runtime/pkg/tables"
+	"github.com/spf13/cobra"
+	"k8s.io/utils/strings/slices"
+)
+
+func NewJob(c CommandContext) *cobra.Command {
+	cmd := cli.Command(&Job{client: c.ClientFactory}, cobra.Command{
+		Use:     "job [flags] [ACORN_NAME|JOB_NAME...]",
+		Aliases: []string{"jobs"},
+		Example: `
+acorn jobs`,
+		SilenceUsage:      true,
+		Short:             "Manage jobs",
+		ValidArgsFunction: newCompletion(c.ClientFactory, jobsCompletion).complete,
+	})
+	cmd.AddCommand(NewJobRestart(c))
+	return cmd
+}
+
+type Job struct {
+	Quiet  bool   `usage:"Output only names" short:"q"`
+	Output string `usage:"Output format (json, yaml, {{gotemplate}})" short:"o"`
+	client ClientFactory
+}
+
+func (a *Job) Run(cmd *cobra.Command, args []string) error {
+	c, err := a.client.CreateDefault()
+	if err != nil {
+		return err
+	}
+
+	out := table.NewWriter(tables.Job, a.Quiet, a.Output)
+
+	switch len(args) {
+	case 0:
+		// No app or job name supplied, list all job
+		if err := printJobs(cmd.Context(), c, nil, &out); err != nil {
+			return err
+		}
+	case 1:
+		// One app or job name supplied, only list matches
+		app, err := c.AppGet(cmd.Context(), args[0])
+		if err != nil {
+			// See if it's the name of a job instead
+			job, err := c.JobGet(cmd.Context(), args[0])
+			if err != nil {
+				return err
+			}
+			out.Write(job)
+		} else {
+			if err := printJobs(cmd.Context(), c, &client.JobListOptions{App: app.Name}, &out); err != nil {
+				return err
+			}
+		}
+	default:
+		// More than one name supplied, iterate through job and list any that match
+		jobs, err := c.JobList(cmd.Context(), nil)
+		if err != nil {
+			return err
+		}
+		for _, job := range jobs {
+			if slices.Contains(args, job.Name) {
+				out.Write(&job)
+			}
+		}
+	}
```

Comment:
The behavior here seems a bit inconsistent between different arg cardinatilities:
A) Given one name, it can be an app or a job
B) Given > 1 name, they can only be jobs

I'd expect the behavior of A to apply to each argument when multiple are given.

Related: You can get `NotFound` errors with a single argument, but not with multiple.

Reply:
Alright, I pretty dramatically overhauled this logic to be simpler and more comprehensive. If we like these changes I'll also make a follow-up in the container replicas equivalent code, since what I'm changing was originally pulled from there.